### ISSUE TITLE
Format on Save Mode : Modifications causes invalid indentation

### ISSUE THREAD

**Issue by adnaanbheda** [microsoft/vscode#107026]
- VSCode Version: v1.49.1
- OS Version: Windows 10 2004

Steps to Reproduce:

1. Change Format On Save Mode to modifications
2. Edit some code and save the file

![image](https://user-images.githubusercontent.com/25040547/93613432-7ba27b00-f9ee-11ea-84e3-18580249bd45.png)

Does this issue occur when all extensions are disabled?: Yes/No
Yes

**Comment by oclockvn** [comment]
in my case, the html block is moved to the left side

![image](https://user-images.githubusercontent.com/3783976/93711719-bdf8c300-fb7a-11ea-9538-4fa4f1a37fe4.png)

**Comment by jrieken** [comment]
/needsMoreInfo

Please update this issue after reading https://github.com/Microsoft/vscode/wiki/Submitting-Bugs-and-Suggestions#writing-good-bug-reports-and-feature-requests. Thanks.

**Comment by tcelestino** [comment]
@jrieken, this is happens when `editor.formatOnSaveMode` is set as `modifications`. I've changed the settings to `file` format on save has worked again

**Comment by jrieken** [comment]
And what formatter extension do you use? and what language is this?

**Comment by tcelestino** [comment]
@jrieken I'm using Prettier. The language: Javascript (and Typescript).

My settings:
```json
"[javascript]": {
	"editor.formatOnSave": true,
    "editor.defaultFormatter": "esbenp.prettier-vscode"
},
"[typescript]": {
	"editor.formatOnSave": true,
    "editor.defaultFormatter": "esbenp.prettier-vscode"
},
"[typescriptreact]": {
    "editor.formatOnSave": true,
    "editor.defaultFormatter": "esbenp.prettier-vscode"
},
"[javascriptreact]": {
	"editor.formatOnSave": true,
    "editor.defaultFormatter": "esbenp.prettier-vscode"
},
```

**Comment by jrieken** [comment]
Thanks for clarifying. Please file this issue against the prettier extensions, since its the extension that determines that changes, including indentation

**Comment by cp79shark** [comment]
@tcelestino I found it's because I'm using an old version of Prettier in my project (1.19.1). Updating to the latest 2.1.2 fixed the same issue with that setting.

**Comment by localjo** [comment]
I had a similar issue when using the new "modifications" option for Format On Save Mode with Prettier as the formatter. Modified lines were formatted incorrectly, specifically, I was seeing lines being indented way too far when modifying the inside of a block. They would be pushed to the right of the opening bracket.

If I would edit the contents of a block like this:

```
if (someCondition === true) {
  console.log('Condition is true!');
}
```

It would get formatted something like this:

```
if (someCondition === true) {
                              console.log('Condition is still true!');
                            }
```

It appears that the root of the issue for me was that older versions of Prettier would handle indentation incorrectly with `rangeStart` and `rangeEnd`. In most cases, the recommended fix is to upgrade Prettier in the project to the latest with something like `npm install prettier@2.1.2 --save-exact --save-dev` or `npm install prettier@latest --save-dev`. In my case, I decided to use a global installation of Prettier@2.1.2 and set the Prettier Path setting in the Prettier plugin to the global Prettier path and that solved the problem.

TL;DR The VS Code "modifications" option for Format On Save Mode doesn't work with older versions of Prettier.
